test/quantization/test_observer.py

```diff
@@ -199,7 +200,6 @@ def test_linear_observer_tensor(self, observe_weight: bool):
             input_scale.item(),
             max_val / max_fp8,
         )
-        self.assertIsNotNone(input_zero_point)
```

Comment:
is there a change of behavior when you change zero_point_domain for None to ZeroPointDomain.NONE?

Reply:
I see, so what is the meaning of `zero_point_domain` == None before?

Reply:
Some APIs were creating a `None` `zero_point` when `zero_point_domain` `ZeroPointDomain.NONE` or `None` was used, while `choose_qparams_affine` was not.

Reply:
Hi @jerryzh168, is it possible that some torchao users' code may be expecting a non-None `zero_point` with zero_point_domain `ZeroPointDomain.NONE`/None, making this change BC-breaking for them? Thanks!

Reply:
I think that most usages of this function are internal to torchao so that its okay to BC break, you can add the label just to be sure

Reply:
Thanks for your advice, @drisspg!
Could you please help add such a label as GitHub isn't displaying an option to me for adding it? Thanks!

Reply:
I added a bc-breaking label, please also write a bc-breaking note similar to https://github.com/pytorch/ao/pull/1049

Reply:
Thanks again, @jerryzh168! I added a note & rebased the PR.